src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonUtilities.Schema.cs

```diff
@@ -302,7 +302,7 @@ JsonNode TransformSchemaNode(JsonSchemaExporterContext ctx, JsonNode schema)
                     _ = objSchema.TryGetPropertyValue(RequiredPropertyName, out JsonNode? required);
                     if (required is not JsonArray { } requiredArray || requiredArray.Count != propertiesObj.Count)
                     {
-                        requiredArray = [.. propertiesObj.Select(prop => prop.Key)];
+                        requiredArray = [.. propertiesObj.Select(prop => (JsonNode)prop.Key)];
```

Comment:
Casting to `JsonNode` results in the non-generic `JsonArray.Add` method being used which is linker-safe.

Reply:
Why is the non-generic safe but the generic isn't? i.e. what is the explicit cast to JsonNode doing that's safe and that can't be done in the generic Add?

Reply:
Can we add a quick comment explaining why the cast is necessary?

Reply:
> Why is the non-generic safe but the generic isn't? i.e. what is the explicit cast to JsonNode doing that's safe and that can't be done in the generic Add?

The generic `Add` method encapsulates arbitrary .NET values and uses reflection to lazily serialize them to JSON. The cast uses the explicit conversions from known primitives to `JsonNode` which doesn't suffer from the same problem.

> Can we add a quick comment explaining why the cast is necessary?

If you search for casts to `JsonNode` in this source file, you'll see that this pattern is pervasive both here and generally any AOT compatible library that works with `JsonNode`. It sucks, but that's what got shipped before trimmer annotations became a design concern.